python/ray/dag/dag_node.py

```diff
@@ -103,9 +107,20 @@ async def get_object_refs_from_last_execute(self) -> Dict[str, Any]:
     def clear_cache(self):
         self.cache_from_last_execute = {}
 
+    def compiled(self) -> Tuple[ray.ObjectRef]:
+        if self._compiled_dag is None:
+            self._compiled_dag = build_compiled_dag(self)
+
+        return self._compiled_dag.compiled()
+
     def execute(
-        self, *args, _ray_cache_refs: bool = False, **kwargs
-    ) -> Union[ray.ObjectRef, ray.actor.ActorHandle]:
+        self,
+        *args,
+        _ray_cache_refs: bool = False,
+        _ray_cache_actors: bool = True,
+        compiled: bool = False,
```

Comment:
Shall we make an `experimental_compile()` method that returns a compiled node type instead of adding an arg here? I think the arg is a bit confusing since it's not clear that we cache the compiled DAG on the first interaction with execute.